### ISSUE TITLE
Error when exporting to TensorRT

### ISSUE THREAD

**Issue by Rhierarch** [ultralytics/ultralytics#6861]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Export

### Bug

When I tried to export my best.pt(pretrained from yolov8n.pt) to TensorRT,it raised out error:
"
WARNING ⚠️ TensorRT requires GPU export, automatically assigning device=0
Ultralytics YOLOv8.0.222 🚀 Python-3.10.2 torch-2.1.0+cu121 CUDA:0 (NVIDIA GeForce RTX 4060, 8188MiB)
YOLOv8n-pose summary (fused): 187 layers, 3289964 parameters, 0 gradients, 9.2 GFLOPs

PyTorch: starting from 'yolo\yolov8n-pose.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 56, 8400) (6.5 MB)

ONNX: starting export with onnx 1.15.0 opset 17...

Process finished with exit code -1073741819 (0xC0000005)
"
 And my engine haven't been saved.

### Environment

Python=3.10.2
torch-2.1.0+cu121 CUDA:0 (NVIDIA GeForce RTX 4060, 8188MiB)
cuda=12.3 
cudnn=8.9.1
TensorRT=8.6.1.6
ultralytics=8.0.222
numpy =1.23.4
onnx =1.15.0
onnxconverter-common=1.14.0
onnxmltools =1.11.2
onnxruntime-gpu =1.16.1
onnxsim = 0.4.35

### Minimal Reproducible Example

from ultralytics import YOLO
model = YOLO('yolo/yolov8n-pose.pt')
model.export(format='TensorRT')

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@Rhierarch this error code generally indicates a memory access violation, which may not be directly related to YOLOv8 itself but could be arising from the environment setup or an issue during the ONNX export process, or while converting the model with TensorRT.

Make sure that your system has sufficient memory and storage available. Sometimes, rebooting your machine can clear any transient issues that may cause such errors.

Also, validate that your environment meets all the requirements for exporting a model to TensorRT, including compatible versions of PyTorch, ONNX, CUDA, and TensorRT itself.

If the issue persists, consider exporting the model to ONNX format first using `model.export(format='onnx')` to see if the problem is specific to TensorRT conversion. If the ONNX export succeeds, use the TensorRT command line tools to convert the ONNX model to TensorRT, which may provide more verbose error messaging.

If you're still experiencing trouble, please open a new issue providing the details of the steps taken and any additional error messages or logs. Your willingness to help by submitting a PR is much appreciated! 🙌

**Comment by Rhierarch** [comment]
@glenn-jocher  I quickly tried the same export script on the A40 and found that it worked. I then tried again on the original 4060 and opened the device performance console. For a moment, the memory usage shot up and the export process quit.

Therefore, insufficient video memory should cause the export failure. When I added workspace=1, the export still failed, indicating that this parameter only limits the video memory used by the exported model in the inference process, but does not limit the export process.

So how do I implement the export function without replacing the device?

**Comment by glenn-jocher** [comment]
@Rhierarch It appears that the issue is indeed related to GPU memory constraints during the export process on your RTX 4060. The `workspace` parameter does indeed only affect inference memory, not the export procedure.

To address this without replacing your device, consider the following suggestions:

1. Reduce the batch size during export if it's larger than 1.
2. Close any unnecessary applications or processes that might be consuming GPU memory.
3. Try reducing the model's input resolution temporarily for the export process.

If these steps do not resolve the issue, it may be necessary to perform the export on a system with a more capable GPU that has more memory. After exporting, you can transfer the TensorRT model back to your RTX 4060 for inference. 🤞
